src/services/fluence-aurora-connector/modules/fluence_aurora_connector/src/curl.rs

```diff
@@ -0,0 +1,79 @@
+use marine_rs_sdk::marine;
+use marine_rs_sdk::MountedBinaryResult;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::jsonrpc::request::RequestError;
+use crate::jsonrpc::request::RequestError::{CurlError, OtherError, ParseError, RpcSerializeError};
+use crate::jsonrpc::{JsonRpcReq, JsonRpcResp};
+
+#[marine]
+#[link(wasm_import_module = "curl_adapter")]
+extern "C" {
+    pub fn curl_request(cmd: Vec<String>) -> MountedBinaryResult;
+}
+
+#[rustfmt::skip]
+fn curl_params(url: String, data: String) -> Vec<String> {
+    let params = vec![
+        url.as_str(),
+        // To avoid unnecessary data in stderr
+        "--no-progress-meter",
+        "-X", "POST",
+        "-H", "Content-Type: application/json",
+        // To avoid hanging on try to connect
+        // TODO: what the best timeout?
+        "--connect-timeout", "5",
+        // Do not try to reconnect, just make another call
+        "--retry", "0",
+        "--data", data.as_str(),
+    ];
+
+    params.into_iter().map(String::from).collect::<_>()
+}
+
+pub fn curl(url: String, data: String) -> Result<String, RequestError> {
+    let params = curl_params(url, data);
+    let result = curl_request(params);
+
+    match result.into_std() {
+        None => {
+            return Err(OtherError(
+                "curl output is not a valid UTF-8 string".to_string(),
+            ))
+        }
+        Some(Err(err)) => return Err(CurlError(err)),
+        Some(Ok(result)) => Ok(result),
+    }
+}
+
+pub fn send_jsonrpc<Req: Serialize, Resp: DeserializeOwned>(
+    url: String,
+    req: JsonRpcReq<Req>,
+) -> Result<JsonRpcResp<Resp>, RequestError> {
+    let req = serde_json::to_string(&req).map_err(RpcSerializeError)?;
+    log::debug!("json rpc request: {}", req);
+    let result = curl(url, req)?;
+    log::debug!("json rpc response: {}", result);
```

Comment:
we have here a lot of debug output, but not in the `_batch` function

Mb keep it consistent and remove/add debug logs where needed